v2/tools/asoctl/cmd/import_azure_resource.go

```diff
@@ -33,12 +45,46 @@ func newImportAzureResourceCommand() *cobra.Command {
 	return cmd
 }
 
-// TODO: export resource logic goes here
-func exportResource(armID string, output *string) error {
-	klog.Infof("armID : %s", armID)
+// importAzureResource imports an ARM resource and writes the YAML to stdout or a file
+func importAzureResource(ctx context.Context, armIDs []string, outputPath *string) error {
+	//TODO: Support other clouds
+
+	activeCloud := cloud.AzurePublic
+	creds, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return errors.Wrap(err, "unable to get default azure credential")
+	}
+
+	options := &genericarmclient.GenericClientOptions{
+		UserAgent: "asoctl/" + version.BuildVersion,
+	}
+
+	client, err := genericarmclient.NewGenericClient(activeCloud, creds, options)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create ARM client")
+	}
+
+	importer := importing.NewResourceImporter(api.CreateScheme(), client)
+	for _, armID := range armIDs {
+		importer.AddARMID(armID)
+	}
+
+	result, err := importer.Import(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to import resources")
+	}
 
-	if output != nil && *output != "" {
-		klog.Infof("output : %s", *output)
+	if outputPath == nil || *outputPath == "" {
```

Comment:
minor: You can use the new (added in #2817) `to.Value` here to get `if to.Value(outputPath) == ""` if you wanted.

But fine as is too.